ledger-service/lf-value-json/specification.rst

```diff
@@ -0,0 +1,433 @@
+.. Copyright (c) 2019 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+DAML-LF JSON Encoding
+=====================
+
+We describe how to decode and encode DAML-LF values as JSON. For each
+DAML-LF type we explain what JSON inputs we accept (decoding), and what
+JSON output we produce (encoding).
+
+The output format is parameterized by two flags::
+
+    encodeDecimalAsString: boolean
+    encodeInt64AsString: boolean
+
+The suggested defaults for both of these flags is false.
```

Comment:
why `false`? with `false` data loss might occur.
Plus we currently set both flags to `true`:
https://github.com/digital-asset/daml/blob/b4c00727b4ddbaf2da2b015bfda75b762d4e6f6e/ledger-service/lf-value-json/src/main/scala/com/digitalasset/daml/lf/value/json/ApiCodecCompressed.scala#L312-L313

Reply:
> why `false`? with `false` data loss might occur.

Like I wrote elsewhere, this is un`false`ifiable 🙂 

This is just a copy of the original doc, and unless you're interacting with a peer written in a particular programming language, numeric numbers are the better choice. Our own defaults are because of our likelihood of interacting with that particular programming language.

Reply:
In this case I would add a line or two explaining the implications of the recommended settings. You talk about this below, but it is not clear that to avoid data loss in JavaScript client you need to set those flags to `true`.

Reply:
Sure. 2f993a9014